enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/ServiceDiscoveryEnricher.java

```diff
@@ -107,22 +118,42 @@ public ServiceDiscoveryEnricher(EnricherContext buildContext) {
 
         if (kind == Kind.SERVICE) {
 
-            tryCamelDSLProject();
+            if (!tried) {
+                tryCamelDSLProject();
+                //try more project types in future, last match wins
+                tried = true;
+            }
 
             if (discoverable != null) {
                 Map<String, String> labels = new HashMap<>();
-                String labelName = PREFIX + DISCOVERABLE;
+                String labelName = PREFIX;
                 labels.put(labelName, getConfig(Config.discoverable, discoverable));
-                if (log.isVerboseEnabled()) {
-                    log.verbose("Add %s label: %s=%s", PREFIX, 
-                            labelName, labels.get(labelName));
-                }
+                log.info("Add %s label: \"%s\" : \"%s\"", PREFIX, 
+                        labelName, labels.get(labelName));
                 return labels;
             }
         }
         return super.getLabels(kind);
     }
 
+    @Override
+    public void addMissingResources(final KubernetesListBuilder listBuilder) {
+        listBuilder.accept(new TypedVisitor<ServiceBuilder>() {
+
+            @Override
+            public void visit(ServiceBuilder serviceBuilder) {
+                
```

Comment:
How are you sure that the service given here is the one with the annotations found ? You can configure many services, e.g. when configured manually via fragments. Not sure how this works with you stateful enricher.

You can easily put the wrong port in the annotation when e.g. an additional service with different ports comes last  in the visitor.

Reply:
I'm taking the first port - which it logs out. TBH I was against specifying the port at all but here we are. The user can always override with the correct port, but 90% of the time this will probably work just fine. There is no way of knowing what the correct port is for the enricher.

Reply:
As already commented, it's currently the **other way round**: The autodetection always overrides a specified port.

Tbh, I would do *all* the handling (adding labels and annotations) in the visitor. That way you can avoid the intermediate state handover with enricher member variables and you can deal with multiple services.

Reply:
If you can't specify the port you can't annotate headless services, so adding a port makes sense for this case.

Reply:
So you're saying that defaultValue is used over the one given by the config here?

getConfig(Configs.Key key, String defaultVal)